sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -70,6 +70,14 @@ private[hive] class SparkExecuteStatementOperation(
     }
   }
 
+  def cancelStatement(): Unit = {
+    // RDDs will be cleaned automatically upon garbage collection.
+    logDebug(s"CANCELING $statementId")
+    cleanup(OperationState.CANCELED)
+    sqlContext.sparkContext.clearJobGroup()
+    HiveThriftServer2.listener.onStatementCanceled(statementId)
+  }
+
```

Comment:
In current version, we can't just override cancel() method, so I add a new special one for SparkExecuteStatementOperation.